core/server/worker/src/main/java/alluxio/worker/block/AsyncCacheRequestManager.java

```diff
@@ -107,6 +108,17 @@ public void submitRequest(Protocol.AsyncCacheRequest request) {
    */
   private boolean cacheBlockFromUfs(long blockId, long blockSize,
       Protocol.OpenUfsBlockOptions openUfsBlockOptions) {
+    // Check if the block has already been cached on this worker
+    long lockId = mBlockWorker.lockBlockNoException(Sessions.ASYNC_CACHE_SESSION_ID, blockId);
+    if (lockId != BlockLockManager.INVALID_LOCK_ID) {
+      try {
+        mBlockWorker.unlockBlock(lockId);
```

Comment:
You could use the other api which does not throw exception?

Reply:
do you mean 
```
// TODO(calvin): Remove when lock and reads are separate operations.
boolean unlockBlock(long sessionId, long blockId);
```
I thought we are removing this API?

Reply:
mm good point